integration-tests/test_gcs.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+import io
+import os
+
+from google.cloud import storage
+
+import smart_open
+
+_GCS_BUCKET = os.environ.get('SO_GCS_BUCKET')
+_GCS_URL = 'gs://' + _GCS_BUCKET
+assert _GCS_BUCKET is not None, 'please set the SO_GCS_BUCKET environment variable'
+
+
+def initialize_bucket():
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(_GCS_BUCKET)
+    blobs = bucket.list_blobs()
+    for blob in blobs:
+        blob.delete()
+
+
+def write_read(key, content, write_mode, read_mode, encoding=None, **kwargs):
+    with smart_open.smart_open(key, write_mode, encoding=encoding, **kwargs) as fout:
```

Comment:
Please use the `open` function, here and everywhere. The `smart_open` function is deprecated.